tests/test.py

```diff
@@ -1357,6 +1358,76 @@ def test_rule_expressions_cuckooreport_context(self):
         result = rule.evaluate(sample)
         self.assertEqual(result.result, Result.bad)
 
+    def test_rule_expressions_cortexreport_fileinfoanalyzerreport_context(self):
+        """ Test generic rule cortexreport.FileInfoAnalyzerReport context """
+        config = '''[expressions]
+                expression.2  : cortexreport.FileInfoReport.md5sum == '78576e618aff135f320601e49bd8fe7e' -> good
+                expression.4  : '.day.um' in cortexreport.FileInfoReport.domain_artifacts -> unknown
+                expression.6  : '8.8.8.8' in cortexreport.FileInfoReport.ip_artifacts -> bad
+        '''
+        rule = ExpressionRule(CreatingConfigParser(config), None)
+
+        factory = CreatingSampleFactory(
+            base_dir=None, job_hash_regex=None,
+            keep_mail_data=False, processing_info_dir=None)
+
+        sample = factory.make_sample(os.path.join(
+            self.office_data_dir, 'blank.doc'))
+
+        report = {
+            "summary": {},
+            "full": {
+                "results": [{
+                    "submodule_name": "Basic properties",
+                    "results": [
+                        {
+                            "submodule_section_header": "Hashes",
+                            "submodule_section_content": {
+                                "md5": "78576e618aff135f320601e49bd8fe7e",
+                            }
+                        },
+                    ]
+                }]
+            },
+            "success": False,
+            "artifacts": [],
+        }
+        artifacts = [
+            {
+                "data": "oh.my.day.um",
+                "dataType": "domain",
+            },
+            {
+                "data": "8.8.8.8",
+                "dataType": "ip",
+            }
+        ]
+
+        cortexreport = CortexReport()
+        cortexreport.register_report(FileInfoAnalyzer, report)
+
+        sample.register_cortex_report(cortexreport)
+        result = rule.evaluate(sample)
+        self.assertEqual(result.result, Result.good)
+
+        report["full"]["results"][0]["results"][0]["submodule_section_content"]["md5"] = "A"*32
+        report["artifacts"].append(artifacts[1])
+        sample.register_cortex_report(cortexreport)
+        result = rule.evaluate(sample)
+        self.assertEqual(result.result, Result.bad)
+
+        report["artifacts"].pop(0)
+        report["artifacts"].append(artifacts[0])
```

Comment:
This popping and appending of list elements needs a lot of brain cycles to follow. Couldn't we just do:
```
        report["artifacts"] = [
            {
                "data": "oh.my.day.um",
                "dataType": "domain",
        ]
```
at the first test and then
```
        report["artifacts"] = [
            {
                "data": "8.8.8.8",
                "dataType": "ip",
            }
        ]
```
and be done with it? I just realise that I switched the `[0]` and `[1]` in order which is exactly what I mean regarding required brainpower to get this right. ;)